### ISSUE TITLE
Improve support for arrays of values and duplicate parameters

### ISSUE THREAD

**Issue by vitaly-t** [vitaly-t/connection-string#43]
See [this](https://stackoverflow.com/questions/11621477/using-duplicate-parameters-in-a-url).

* At the moment, we do not support comma-separated values from a single parameter. This can be improved.
* For duplicate parameters, we should construct an array of values automatically, and not throw an error (#19).

**Usage Example**

Let's say, your connection is a database that implements multi-tenant architecture, and you want your connection string to specify which tenant id-s are to be accessible within this connection, which you can do like this:

```
my-server/database?tenantId=1,2,3,4,5
```

or maybe even like this:

```
my-server/database?tenantId=1&tenantId=2,3&tenantId=4,5
```

then in both cases you will get the exact same object, as below:

```js
ConnectionString {
  hosts: [ { name: 'my-server', type: 'domain' } ],
  path: [ 'database' ],
  params: { tenantId: [ '1', '2', '3', '4', '5' ] }
}
```
i.e. all the values are correctly accumulated into an array of strings.

**Comment by vitaly-t** [comment]
After this change, the following applies...

* Comma-separated values turn the parameter into array of values automatically:

```js
const cs = new ConnectionString('?one=1&two=1,2,3');
//=> params: {one: '1', two: ['1', '2', '3']}
```

* Repeated parameters no longer throw an error. Instead, values are accumulated into an array:

```js
const cs = new ConnectionString('?one=1&one=2,3,4&one=5');
//=> params: {one: ['1', '2', '3', '4', '5']}
```

* Whenever a parameter contains an array, the value is serialized into a CSV string:

```js
const cs = new ConnectionString(null, {
    params: {
        one: [1, 2, 3]
    }
});
cs.toString(); //=> ?one=1,2,3
```

**Comment by vitaly-t** [comment]
Released in version [4.3.0](https://github.com/vitaly-t/connection-string/releases/tag/4.3.0).

**Comment by vitaly-t** [comment]
A problem has been found in this implementation, which made `%2C`-encoded `,` treated as a separator.

This should not be the case, only `,` should be treated as a separator, while `%2C` is just part of text-value.

This needs to be fixed.

**Comment by vitaly-t** [comment]
Fixed and released in version [4.3.1](https://github.com/vitaly-t/connection-string/releases/tag/4.3.1).

**Comment by vitaly-t** [comment]
Summary for this was added with FAQ section:

* [How to pass arrays of values?](https://github.com/vitaly-t/connection-string/wiki/FAQ#how-to-pass-arrays-of-values)
